### PR TITLE
Use native array_replace_recursive

### PR DESCRIPTION
**Pull request by Alkarex**
Since PHP 5.3+, there is a native equivalent to the SimplePie Misc function to merge arrays: https://php.net/array-replace-recursive
Improvement of https://github.com/simplepie/simplepie/pull/268

### CODE REVIEW

**Comment by jtojnar** [comment]
Note that the functions are not equivalent. For example:

```php
<?php

require_once __DIR__ . '/vendor/autoload.php';

use SimplePie\Misc;

var_dump(Misc::array_merge_recursive(['a' => 'b'], ['a' => ['b']]));
var_dump(array_replace_recursive(['a' => 'b'], ['a' => ['b']]));
```

prints

```php
array(1) {
  ["a"]=>
  string(1) "b"
}
array(1) {
  ["a"]=>
  array(1) {
    [0]=>
    string(1) "b"
  }
}
```

Although the behaviour of `array_replace_recursive` makes more sense to me and there probably won’t be arrays in the values we are merging.

**Comment by Alkarex** [comment]
Good catch :-) I do not think this was the intended behaviour of `Misc::array_merge_recursive()` and was probably an implementation bug (cf. https://github.com/simplepie/simplepie/pull/268 ). The native `array_replace_recursive()` indeed handles arrays better.

File: src/Misc.php [hunk 1]
```diff
@@ -199,17 +199,12 @@ public static function fix_protocol($url, $http = 1)
         return $url;
     }
 
+    /**
+     * @deprecated since SimplePie 1.7.1, use PHP native array_replace_recursive() instead.
+     */
     public static function array_merge_recursive($array1, $array2)
     {
-        foreach ($array2 as $key => $value) {
```
Comment (Art4) [inline src/Misc.php:204]: For BC reasons I would leave the Misc::array_merge_recursive() method untouched. I think that it is enough if we mark it as deprecated and no longer use it internally.

Comment (Alkarex) [reply src/Misc.php:204]: Done

**Review by Art4 — APPROVED** [review]

File: src/Misc.php [hunk 2]
```diff
@@ -199,6 +199,9 @@ public static function fix_protocol($url, $http = 1)
         return $url;
     }
 
+    /**
+     * @deprecated since SimplePie 1.7.1, use PHP native array_replace_recursive() instead.
```
Comment (jtojnar) [inline src/Misc.php:203]: Maybe also mention that the behaviour is slightly different in case someone really heeds this recommendation?

Comment (Art4) [reply src/Misc.php:203]: I don't think that's necessary. The developer must always check if and how one can use an offered alternative.

**Pull request merged**